test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -14043,6 +14043,21 @@ BOOST_AUTO_TEST_CASE(flipping_sign_tests)
 	ABI_CHECK(callContractFunction("f()"), encodeArgs(true));
 }
 
+BOOST_AUTO_TEST_CASE(external_public_override)
+{
+	char const* sourceCode = R"(
```

Comment:
I don't think this test actually tests the override, does it? In the test we can just remove ``A`` entirely and it will do just the same and not fail - to actually test that the override works we would need, e.g., a contract ``C`` that creates an instance of ``B`` and passes it somewhere as an ``A`` and then calls ``f`` on it, still resulting in ``B``'s ``f`` being called, wouldn't we?

Reply:
Maybe not too important, though - I'm pretty sure that it'll work either way.

Reply:
That's why I did not put a test there in the first place (but still good to have it) - if you call an external function, the type of the contract you call it on does not matter, only the function signature. The C++ interface does not know about the types. We could add an external call from within solidity, but well...